### PR TITLE
feat(rpc): `starknet_subscribeEvents`

### PR DESCRIPTION
**Pull request by sistemd**
Closes https://github.com/eqlabs/pathfinder/issues/2216. Includes some refactorings that I thought I was going to need, I didn't really end up needing them but I still kept them around because they make the code slightly better.

### CODE REVIEW

File: crates/rpc/src/jsonrpc/websocket/data.rs [hunk 1]
```diff
@@ -96,6 +102,20 @@ pub(super) enum ResponseEvent {
...
+pub struct EmittedEvent {
+    pub data: Vec<EventData>,
+    pub keys: Vec<EventKey>,
+    pub from_address: ContractAddress,
+    /// [`None`] for pending events.
+    pub block_hash: Option<BlockHash>,
+    /// [`None`] for pending events.
+    pub block_number: Option<BlockNumber>,
+    pub transaction_hash: TransactionHash,
+}
```
[the remark this answers is not in the document]
Comment (kkovaacs) [reply crates/rpc/src/jsonrpc/websocket/data.rs:117]: Yeah, makes sense, thanks!

File: crates/storage/src/connection/event.rs [hunk 2]
```diff
@@ -89,17 +84,82 @@ impl Transaction<'_> {
...
     pub fn events(
         &self,
         filter: &EventFilter,
         max_blocks_to_scan: NonZeroUsize,
         max_uncached_bloom_filters_to_load: NonZeroUsize,
     ) -> Result<PageOfEvents, EventFilterError> {
-        if filter.page_size > PAGE_SIZE_LIMIT {
-            return Err(EventFilterError::PageSizeTooBig(PAGE_SIZE_LIMIT));
-        }
-
```
[the remark this answers is not in the document]
Comment (kkovaacs) [reply crates/storage/src/connection/event.rs:102]: Indeed, thanks! Totally makes sense.

**Review by kkovaacs — APPROVED** [review]
LGTM, thanks!

**Review by CHr15F0x — APPROVED** [review]
Wow, it was a big one :bow:

**Pull request merged**